### ISSUE TITLE
Selector "has tag" actually means "is only that tag"

### ISSUE THREAD

**Issue by catswithbenefits** [mdelobelle/obsidian_supercharged_links#155]
## To reproduce
- Create a note with the tags `[foo, bar]`
- Create a tag selector for `foo`

## Expected behaviour
- the note with the tags `[foo, bar]` gets supercharged

## Actual behaviour
- it doesn't get supercharged
- only notes with the single tag `foo`will get supercharged

Is this a bug or a feature? Since the UI says "Note has tag #foo" I suspect the first. If it behaves as expected I suggest to change the description in the UI.

**Comment by HEmile** [comment]
This has never been an issue for me. What does the note look like? Where do you put the tags?

**Comment by randomsnowflake** [comment]
You are right, it does not appear to be an issue with multiple tags.

The problem is that sometimes notes with one tag are not supercharged correctly. I apply the tags via frontmatter.


Please see the attached screenshots (retracted for privacy) for details. I have not been able to setup a minimal setup that would trigger the issue.

It also only seems to affect the tag "workout_log" - my other tagged notes ("recipes", "book") have styled applied to them correctly.



<img width="2168" alt="not supercharged" src="https://user-images.githubusercontent.com/11420149/199413971-84e842c1-4862-4d86-815d-203dc090354f.png">
<img width="662" alt="settings" src="https://user-images.githubusercontent.com/11420149/199413977-4292edbe-c758-435a-afe6-d0951b676ed3.png">
<img width="2168" alt="supercharged" src="https://user-images.githubusercontent.com/11420149/199413980-f80d6ec9-7ab2-49e5-912b-da2d1bba71ba.png">

**Comment by HEmile** [comment]
Okay this is probably an issue with parsing frontmatter tags. I don't use these so they're not well tested. I'll look into this in the next round!

**Comment by randomsnowflake** [comment]
Yep, putting the tag (with a hashtag) in the content gets rid of the problem. So it's definitly related to frontmatter.
